### PR TITLE
Summary of Current Sinopia implementation

### PR DESCRIPTION
**Pull request by jermnelson**
First pass at high-level documentation of the technologies used in the Sinopia Server stack.

### CODE REVIEW

File: implementation.md [hunk 1]
```diff
@@ -2,4 +2,61 @@
 layout: default
 ---
 
-Work in progress. 
+# Sinopia Technology Implementation
+The Sinopia stack uses [Circle-ci][CI] for continuous integration to
```
Comment (mjgiarlo) [inline implementation.md]: What's `[Circle-ci][CI]`? Looks like you started to create a link and maybe didn't loop back to add in the URL and correct the syntax?

File: implementation.md [hunk 2]
```diff
@@ -2,4 +2,61 @@
...
+validate the code syntax using [eslint](https://eslint.org/) and test the code base using a combination of
+[Jest](https://jestjs.io/), [Enzyme](https://github.com/airbnb/enzyme), and
+[Puppeteer](https://developers.google.com/web/tools/puppeteer/). [Circle-ci][CI] also
```
Comment (mjgiarlo) [inline implementation.md]: Two more of those link-not-link vestiges here.

File: implementation.md [hunk 3]
```diff
@@ -2,4 +2,61 @@
...
+creates [Docker][DOCK] images on Dockerhub that are automatically deployed to Sinopia's AWS
+development environment.
+
+## Sinopia Linked Data Editor
+A hard fork of the Library of Congress [BIBFRAME Editor](https://github.com/lcnetdev/bfe),
+Sinopia's Linked Data Editor is a [node.js][NODE] project that
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 4]
```diff
@@ -2,4 +2,64 @@
...
+uses the open-source javascript library [React](https://reactjs.org/) to build
+the client-side user interface with supporting [node.js][NODE] libraries like
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 5]
```diff
@@ -2,4 +2,64 @@
...
+[Redux](https://redux.js.org/) for managing the data state of the editor.
+
+The Linked Data Editor also implements a [Swagger REST API][SWAG]
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 6]
```diff
@@ -2,4 +2,64 @@
...
+for search and look-up interactions with third parties like the Questioning
+Authority service from Cornell. The Sinopia Linked Data Editor requires a valid
+[JSON Web Tokens (JWT)][JWT] from users that authenticate through
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

Comment (mjgiarlo) [reply implementation.md]: Should `Tokens` here be `Token`?

File: implementation.md [hunk 7]
```diff
@@ -2,4 +2,64 @@
...
+[Amazon Web Service's Cognito](https://aws.amazon.com/cognito/) for create-delete-update
+operations for both resource templates and RDF data.
+
+Sinopia's Linked Data Editor source code repository is hosted on Github at
+[https://github.com/ld4p/sinopia_editor](https://github.com/LD4P/sinopia_profile_editor).
+
+## Sinopia Profile Editor
+Sinopia's Profile Editor is a forked version of Library of Congress
+[Profile Editor](https://github.com/lcnetdev/profile-edit) that allows users to
+build resource templates contained in profiles and save resulting file to the
```
Comment (mjgiarlo) [inline implementation.md]: `save resulting files` or `save the resulting file`?

File: implementation.md [hunk 8]
```diff
@@ -2,4 +2,64 @@
...
+local systems. Minimal refactoring was done to this codebase with new Sinopia
+branding and user interfaces modifications support the use of
```
Comment (mjgiarlo) [inline implementation.md]: `user interfaces` should be `user interface` here, I think.

File: implementation.md [hunk 9]
```diff
@@ -2,4 +2,64 @@
...
+[JSON schema validation](https://json-schema.org/).
+
+Sinopia's Profile Editor source code repository is hosted on Github at
+[https://github.com/LD4P/sinopia_profile_editor](https://github.com/LD4P/sinopia_profile_editor).
+
+## Sinopia Server
+The Sinopia Server includes a Docker environment that uses the linked-data
+platform [Trellis][LDP] as the main technology for managing
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 10]
```diff
@@ -2,4 +2,64 @@
...
+both RDF and non-RDF data. The Sinopia Stack uses the [Postgres](https://www.postgresql.org/)
+relational-database variant of [Trellis][LDP] for the storage layer with an
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 11]
```diff
@@ -2,4 +2,64 @@
...
+[Elasticsearch](https://www.elastic.co/products/elasticsearch) search index for
+discovery and search. Create-update-delete operations in [Trellis][LDP] are
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 12]
```diff
@@ -2,4 +2,64 @@
...
+controlled by incoming [JWTs][JWT] from the Sinopia Linked Data editor or supporting
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 13]
```diff
@@ -2,4 +2,64 @@
...
+tools.
+
+The primary method of interaction between the Sinopia Linked Data Editor and the
+Sinopia Server is through a [Swagger REST API][SWAG] client that is published as
```
Comment (mjgiarlo) [inline implementation.md]: Not a Markdown link.

File: implementation.md [hunk 14]
```diff
@@ -2,4 +2,64 @@
...
+a [npm (node package manager)](http://npmjs.com/) package at
+https://www.npmjs.com/package/sinopia_server for importing into the
```
Comment (mjgiarlo) [inline implementation.md]: Would be nice to make this a link.

Comment (michelleif) [inline implementation.md]: "for importing into the Sinopia Linked Data Editor"--don't understand that

File: implementation.md (hunk 1 above)
Comment (michelleif) [reply implementation.md]: and Circle-CI should be with capital CI

File: implementation.md [hunk 15]
```diff
@@ -2,4 +2,64 @@
 layout: default
 ---
 
-Work in progress. 
+# Sinopia Technology Implementation
+The Sinopia stack uses [Circle-ci][CI] for continuous integration to
+validate the code syntax using [eslint](https://eslint.org/) and test the code base using a combination of
+[Jest](https://jestjs.io/), [Enzyme](https://github.com/airbnb/enzyme), and
+[Puppeteer](https://developers.google.com/web/tools/puppeteer/). [Circle-ci][CI] also
```
Comment (michelleif) [inline implementation.md]: Circle-CI here too (capitalize CI)

File: implementation.md (hunk 8 above)
Comment (michelleif) [reply implementation.md]: add comma after "branding" to make it easier to read

File: implementation.md [hunk 16]
```diff
@@ -2,4 +2,64 @@
...
+branding and user interfaces modifications support the use of
+[JSON schema validation](https://json-schema.org/).
+
+Sinopia's Profile Editor source code repository is hosted on Github at
+[https://github.com/LD4P/sinopia_profile_editor](https://github.com/LD4P/sinopia_profile_editor).
+
+## Sinopia Server
+The Sinopia Server includes a Docker environment that uses the linked-data
+platform [Trellis][LDP] as the main technology for managing
+both RDF and non-RDF data. The Sinopia Stack uses the [Postgres](https://www.postgresql.org/)
```
Comment (michelleif) [inline implementation.md]: Stack was stack above. should be consistent

File: implementation.md (hunk 1 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 3 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 5 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 10 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 4 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 12 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 14 above)
Comment (jermnelson) [reply implementation.md]: Done in latest commit.

Comment (jermnelson) [reply implementation.md]: Removed in latest commit; was trying to express how we actual use this node.js package in the Editor's source code.

File: implementation.md (hunk 15 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 16 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 6 above)
Comment (jermnelson) [reply implementation.md]: It is plural at the https://jwt.io/ website; you're right, in this context should be singular. Fixed in latest commit.

File: implementation.md (hunk 7 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 8 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 9 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 2 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 11 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

File: implementation.md (hunk 13 above)
Comment (jermnelson) [reply implementation.md]: Fixed in latest commit.

**Comment by mjgiarlo** [comment]
@jermnelson I'm still seeing all the busted markdown links.

**Comment by jermnelson** [comment]
> @jermnelson I'm still seeing all the busted markdown links.

Hmm, on the ec6261c78a2c3caba05dbf535ded619055184d8b commit all of the links except one are working for me, I checked the markdown rendering to html these three ways :

- 1. Through Github at https://github.com/LD4P/sinopia/blob/implementation-details/implementation.md
- 2. Running Jekyll locally and going to http://localhost:4000/sinopia/implementation
- 3. Running `pandoc -i implementation.md -o implementation.html`. 

I did notice that the Library of Congress Bibframe Editor syntax was incorrect and I'll push up a fix.

**Comment by mjgiarlo** [comment]
@jermnelson :speech_balloon: 
> 
> Hmm, on the [ec6261c](https://github.com/LD4P/sinopia/commit/ec6261c78a2c3caba05dbf535ded619055184d8b) commit all of the links except one are working for me, I checked the markdown rendering to html these three ways :
> 
> * 1. Through Github at [/implementation.md@`implementation-details`](https://github.com/LD4P/sinopia/blob/implementation-details/implementation.md)
> * 1. Running Jekyll locally and going to http://localhost:4000/sinopia/implementation
> * 1. Running `pandoc -i implementation.md -o implementation.html`.
 
OK, the "preview changes" view in GitHub is now rendering fine. Looks great. :slightly_smiling_face:

File: implementation.md [hunk 17]
```diff
@@ -2,4 +2,66 @@
 layout: default
```
Comment (jgreben) [inline implementation.md]: Is this supposed to be here? It looks strange in the rendered view...

Comment (jgreben) [reply implementation.md]: Otherwise, LGTM

Comment (mjgiarlo) [reply implementation.md]: Yeah, this is [Jekyll front matter](https://jekyllrb.com/docs/front-matter/).

**Comment by ndushay** [comment]
@jermnelson I'd say you've gotten enough eyeballs on this and can merge it yourself;  I'm hesitating to merge b/c if there's something weird with github pages after deploy, you would be better placed to address it, I think.

**Pull request merged**